protocol/proto/roundtrip_test.go

```diff
@@ -149,6 +150,39 @@ func TestDecodeMessageLargerThanMaxSize(t *testing.T) {
 	require.Contains(t, err.Error(), "larger than maximum supported size")
 }
 
+func TestDecodeReset(t *testing.T) {
```

Comment:
nit: can remove this test, there is already one test for this.

Reply:
There used to be one for EncoderDecoder. I removed that one, and added this one for Decoder only.

Reply:
There is already one for resetting the reader in decoder [here](https://github.com/m3db/m3msg/pull/52/files/68f05ee5979c7a7614d71ec34275595a2b0a2613#diff-86b4f7b5f1e92cec58cef06957a702cfR95)